packages/jest-core/src/__tests__/watch-file-changes.test.ts

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+'use strict';
+
+import path from 'path';
+import fs from 'fs';
+import {JestHook} from 'jest-watcher';
+import Runtime from 'jest-runtime';
+import {normalize} from 'jest-config';
+import HasteMap from 'jest-haste-map';
+import rimraf from 'rimraf';
+import {AggregatedResult} from '@jest/test-result';
+
+describe('Watch mode flows with changed files', () => {
+  let watch: any;
+  let pipe: NodeJS.ReadStream;
+  let stdin: MockStdin;
+  const fileTargetPath = path.resolve(__dirname, 'lost-file.js');
+  const fileTargetPath2 = path.resolve(__dirname, 'watch-test-fake.test.js');
+  const cacheDirectory = path.resolve(__dirname, `tmp${Math.random()}`);
```

Comment:
Can we move this out of the Jest project dir, e.g. into a directory in `os.tmpdir()`? When running this test in watch mode, it loops infinitely and I think that might fix it

Reply:
`os.tmpdir` doesnt fix it. I'll try to figure it out.

Reply:
Oh sorry, the fileTargetPaths are probably the ones that actually cause the problem, not so much the cacheDirectory

Reply:
Yes, i have already sent patch )

Reply:
@jeysal  Is it possible to re-run ci tests without new commit? There is a strange error in last mac os ci build `in Install dependencies state`.

Reply:
On Azure I don't know of a way sorry. But you can `git commit --amend --no-edit && git push -f` :)